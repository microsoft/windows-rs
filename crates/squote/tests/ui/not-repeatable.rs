use quote::quote;
use std::net::Ipv4Addr;

fn main() {
    let ip = Ipv4Addr::LOCALHOST;
    let _ = quote! { #(#ip)* };
}
