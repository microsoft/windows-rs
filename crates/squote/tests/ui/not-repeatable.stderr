error[E0599]: no method named `quote_into_iter` found for struct `std::net::Ipv4Addr` in the current scope
  --> $DIR/not-repeatable.rs:6:13
   |
6  |       let _ = quote! { #(#ip)* };
   |               ^^^^^^^^^^^^^^^^^^ method not found in `std::net::Ipv4Addr`
   |
   = note: the method `quote_into_iter` exists but the following trait bounds were not satisfied:
           `std::net::Ipv4Addr: std::iter::Iterator`
           which is required by `std::net::Ipv4Addr: quote::__private::ext::RepIteratorExt`
           `&std::net::Ipv4Addr: std::iter::Iterator`
           which is required by `&std::net::Ipv4Addr: quote::__private::ext::RepIteratorExt`
           `std::net::Ipv4Addr: quote::to_tokens::ToTokens`
           which is required by `std::net::Ipv4Addr: quote::__private::ext::RepToTokensExt`
           `&mut std::net::Ipv4Addr: std::iter::Iterator`
           which is required by `&mut std::net::Ipv4Addr: quote::__private::ext::RepIteratorExt`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
